save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt

```diff
@@ -22,10 +23,14 @@ class OrganizationService(
      * @return organization's id, should never return null
      */
     @Suppress("UnsafeCallOnNullableType")
-    fun getOrSaveOrganization(organization: Organization): Pair<Long, OrganizationSaveStatus> {
-        val (organizationId, organizationSaveStatus) = organizationRepository.findByName(organization.name)?.let {
-            Pair(it.id, OrganizationSaveStatus.EXIST)
-        } ?: Pair(organizationRepository.save(organization).id, OrganizationSaveStatus.NEW)
+    @Transactional
+    fun saveOrganization(organization: Organization): Pair<Long, OrganizationSaveStatus> {
+        val (organizationId, organizationSaveStatus) = if (organizationRepository.validateOrganizationName(organization.name) == 0L) {
```

Comment:
You have already added a primaryKey to this field, now you are double checking for a name lookup. Can use a _try-catch_ in this case?

Reply:
Ok, done. Take a look pls